From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Tue, 29 Jul 2025 15:41:41 +0000
Subject: [PATCH] Restore local password manager UI

Adapted from GrapheneOS Vanadium patch 0211-Restore-local-password-manager-UI.patch for the Vandium browser port.

Revert "[passwords/android] Remove unused PasswordSettings.java"

This reverts commit 3662471ee9fabd6d1777b1d5316f0b9eede0f115.
---
 chrome/android/chrome_java_resources.gni      |   2 +
 chrome/android/chrome_java_sources.gni        |   1 +
 .../java/res/layout/password_no_result.xml    |  35 +
 ...e_password_preferences_action_bar_menu.xml |  29 +
 .../android/java/res/xml/main_preferences.xml |   1 +
 .../java/res/xml/main_preferences_legacy.xml  |   1 +
 .../settings/PasswordSettings.java            | 618 ++++++++++++++++++
 .../settings/SettingsNavigationImpl.java      |   4 +
 chrome/android/javatests/BUILD.gn             |   4 +
 .../settings/PasswordSettingsSearchTest.java  | 534 +++++++++++++++
 .../settings/PasswordSettingsTest.java        | 383 +++++++++++
 .../settings/PasswordSettingsTestHelper.java  | 193 ++++++
 .../settings/PasswordViewingTypeTest.java     | 107 +++
 .../settings/MainSettingsFragmentTest.java    |   3 +-
 chrome/browser/BUILD.gn                       |   2 +
 .../browser/password_manager/android/BUILD.gn |  10 +
 .../password_manager/settings/ExportFlow.java | 606 +++++++++++++++++
 .../settings/ExportFlowInterface.java         |  92 +++
 .../settings/PasswordListObserver.java        |  28 +
 .../settings/PasswordManagerHandler.java      |  83 +++
 .../PasswordManagerHandlerProvider.java       | 144 ++++
 .../settings/PasswordUiView.java              | 191 ++++++
 .../settings/ProgressBarDialogFragment.java   |  70 ++
 .../settings/SavedPasswordEntry.java          |  41 ++
 .../settings/ReauthenticationManagerTest.java |   2 +-
 .../android/password_ui_view_android.cc       | 365 +++++++++++
 .../android/password_ui_view_android.h        | 170 +++++
 .../password_ui_view_android_unittest.cc      | 198 ++++++
 .../FakePasswordManagerHandler.java           | 135 ++++
 .../strings/android_chrome_strings.grd        |  51 ++
 ...NDROID_TRUSTED_VAULT_BANNER_LABEL.png.sha1 |   1 +
 ...ULT_BANNER_SUB_LABEL_OFFER_OPT_IN.png.sha1 |   1 +
 ...D_VAULT_BANNER_SUB_LABEL_OPTED_IN.png.sha1 |   1 +
 .../IDS_PASSWORDS_CHECK_DESCRIPTION.png.sha1  |   1 +
 .../IDS_PASSWORD_NO_RESULT.png.sha1           |   1 +
 chrome/test/BUILD.gn                          |   1 +
 .../settings/SettingsNavigation.java          |  11 +-
 37 files changed, 4114 insertions(+), 6 deletions(-)
 create mode 100644 chrome/android/java/res/layout/password_no_result.xml
 create mode 100644 chrome/android/java/res/menu/save_password_preferences_action_bar_menu.xml
 create mode 100644 chrome/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordSettings.java
 create mode 100644 chrome/android/javatests/src/org/chromium/chrome/browser/password_manager/settings/PasswordSettingsSearchTest.java
 create mode 100644 chrome/android/javatests/src/org/chromium/chrome/browser/password_manager/settings/PasswordSettingsTest.java
 create mode 100644 chrome/android/javatests/src/org/chromium/chrome/browser/password_manager/settings/PasswordSettingsTestHelper.java
 create mode 100644 chrome/android/javatests/src/org/chromium/chrome/browser/password_manager/settings/PasswordViewingTypeTest.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/ExportFlow.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/ExportFlowInterface.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordListObserver.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordManagerHandler.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordManagerHandlerProvider.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordUiView.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/ProgressBarDialogFragment.java
 create mode 100644 chrome/browser/password_manager/android/java/src/org/chromium/chrome/browser/password_manager/settings/SavedPasswordEntry.java
 create mode 100644 chrome/browser/password_manager/android/password_ui_view_android.cc
 create mode 100644 chrome/browser/password_manager/android/password_ui_view_android.h
 create mode 100644 chrome/browser/password_manager/android/password_ui_view_android_unittest.cc
 create mode 100644 chrome/browser/password_manager/android/test_support/java/src/org/chromium/chrome/browser/password_manager/FakePasswordManagerHandler.java
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_ANDROID_TRUSTED_VAULT_BANNER_LABEL.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_ANDROID_TRUSTED_VAULT_BANNER_SUB_LABEL_OFFER_OPT_IN.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_ANDROID_TRUSTED_VAULT_BANNER_SUB_LABEL_OPTED_IN.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORDS_CHECK_DESCRIPTION.png.sha1
 create mode 100644 chrome/browser/ui/android/strings/android_chrome_strings_grd/IDS_PASSWORD_NO_RESULT.png.sha1

diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index f75134432adc4..9d1bb42c023f6 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -3228,6 +3228,8 @@ static_library("browser") {
       "password_manager/android/password_manager_launcher_android.cc",
       "password_manager/android/password_store_bridge.cc",
       "password_manager/android/password_store_bridge.h",
+      "password_manager/android/password_ui_view_android.cc",
+      "password_manager/android/password_ui_view_android.h",
       "password_manager/android/save_update_password_message_delegate.cc",
       "password_manager/android/save_update_password_message_delegate.h",
       "payments/android/service_worker_payment_app_bridge.cc",
diff --git a/chrome/test/BUILD.gn b/chrome/test/BUILD.gn
index 572de58b1cb6b..914241cec18f2 100644
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -7735,6 +7735,7 @@ test("unit_tests") {
       "../browser/page_image_service/android/image_service_bridge_unittest.cc",
       "../browser/page_load_metrics/observers/android_page_load_metrics_observer_unittest.cc",
       "../browser/partnerbookmarks/partner_bookmarks_shim_unittest.cc",
+      "../browser/password_manager/android/password_ui_view_android_unittest.cc",
       "../browser/performance_manager/policies/process_rank_policy_android_unittest.cc",
       "../browser/policy/browser_dm_token_storage_android_unittest.cc",
       "../browser/policy/chrome_browser_cloud_management_controller_android_unittest.cc",
